proto/src/domain/mod.rs

```diff
@@ -31,5 +31,5 @@ where
 /// There are no nullifiers in the genesis block. The value zero is instead used to signal absence
 /// of a value.
 pub fn nullifier_value_to_block_num(value: Word) -> u32 {
```

Comment:
This function is duplicated in a few places in the codebase, such as [here](https://github.com/0xPolygonMiden/miden-node/blob/3c3ebbecbece73b330fe01cfa2b6a41c825c8db1/store/src/state.rs#L482-L484). Ideally we'd only have it in one place - and probably on the `Nullifier` type somehow

Reply:
Thanks, I was thinking about it too and gone to the same conclusion. :) Let me implement it. :)

Reply:
@plafer I've moved it into the `store`.

Reply:
That's fine for now, but I think the ideal solution would be to
1. Use miden-base's [`Nullifier`](https://github.com/0xPolygonMiden/miden-base/blob/9a1b1263b1faf661623d74e00878cf51371b1088/objects/src/notes/nullifier.rs#L23) type in the node
2. Add a `Nullifier::from_block_num(u32) -> Self` method to `Nullifier`, or something like that.

If you agree, can you create an issue for that? Thanks!

Reply:
@plafer no, we use `Nullifier` now for storing nullifier's hash, but these functions for encoding/decoding nullifier's value. Nullifier's value stores just block number, where nullifier was consumed.

Reply:
We can add different type for that, something like `NullifierData`.

Reply:
Ah yes you're right, my bad! Agreed with something like `NullifierData`